pkg/scheduler/schedule_one.go

```diff
@@ -905,6 +915,11 @@ func getAttemptsLabel(p *framework.QueuedPodInfo) string {
 // handleSchedulingFailure records an event for the pod that indicates the
 // pod has failed to schedule. Also, update the pod condition and nominated node name if set.
 func (sched *Scheduler) handleSchedulingFailure(ctx context.Context, fwk framework.Framework, podInfo *framework.QueuedPodInfo, status *framework.Status, nominatingInfo *framework.NominatingInfo, start time.Time) {
+	// Basically, AddUnschedulableIfNotPresent calls DonePod internally.
```

Comment:
there is a very small chance here that `Done` is called after `AddUnschedulableIfNotPresent` and the pod entered the next scheduling cycle.

what we can do is store a boolean `calledDone` that we can set to true when we call `AddUnschedulabelIfNotPresent` and in this case we skip a second call to `Done`.

This is nice because there is only one scenario where we call `AddUnschedulableIfNotPresent`